index.bs

```diff
@@ -1,317 +1,215 @@
+==============================================
 
 <pre class=propdef>
 Name: render-subtree
-Value: normal | visible | invisible && [ skip-activation | skip-viewport-activation ]?
+Value: normal | hidden && frozen?
 Initial: normal
 Inherited: no
 </pre>
 
-<dfn export>invisibly rendered</dfn>
-(unsure if this term is needed,
-it just seemed useful for 'contain-intrinsic-size'’s intro)
-
-
-Default Sizing: the 'contain-intrinsic-size' property {#contain-intrinsic-size}
------------------------------------------------------
-
-<pre class=propdef>
-Name: contain-intrinsic-size
-Value: none | <<length>>{1,2}
-Initial: none
-Inherited: no
-Applies to: elements with [=size containment=]
-</pre>
-
-When an element is [=invisibly rendered=],
-one of the side-effects is that it automatically receives [=size containment=].
-This causes the element to lay out as if it was empty,
-meaning that the author would have to provide an explicit size
-if they wanted the element to take up space on the page
-while it was invisible
-(to reduce the amount of relayout when it is finally rendered,
-ensure scrollbars show a reasonable offset,
-etc).
-
-However, providing an explicit 'width' and 'height' has side effects:
-layout algorithms treat it as a strong signal about what size the element should be
-(overriding things like ''place-self: stretch''),
-and when the element is made visible again,
-the explicit 'width'/'height' has to be removed
-for the element to lay out at its natural content size again.
-
-To avoid these issues,
-but still provide a reasonable default layout size for elements with [=size containment=],
-the 'contain-intrinsic-size' property overrides the normal methods
-of determining an element's [=intrinsic sizes=].
-Values are defined as:
-
-<dl dfn-type=value dfn-for=contain-intrinsic-size>
-	: <dfn>none</dfn>
-	:: No effect;
-		the element is sized as usual.
-
-	: <<length>>{1,2}
-	::
-		If the element is not under [=size containment=],
-		this value has no effect.
-
-		Otherwise,
-		the first <<length>> specifies the [=min-content block size=] and [=max-content block size=]
-		of the element's [=principal box=],
-		and the second specifies the [=min-content inline size=] and [=max-content inline size=]
-		of the element’s [=principal box=].
-
-		If one <<length>> is provided,
-		the second defaults to the same <<length>>.
+<dl dfn-type value dfn-for=render-subtree>
+    : <dfn export>normal</dfn>
+    :: No effect; The element and its subtree are rendered as usual. This
+          provides no extra information or hints to the user-agent.
+
+    : <dfn export>hidden</dfn>
+    :: The element has [=layout containment=] and [=style containment=] applied
+         in addition to any existing containment values.
+
+         Additionally, if the element is <dfn export>off-screen</dfn>, in the
+         sense that it is not intersecting the viewport (with a reasonable
+         user-agent specified margin), then element has [=size containment=]
```

Comment:
Suggest adding a definition called "hidden", which means:
* Size containment
* Not painted
* not hit-tested

Then you can say that when the element is off-screen, it is hidden; otherwise it is not hidden.

Reply:
Oops missed this comment. Can you see if still makes sense in the new spec?

Reply:
I think it would be better to first define "hidden" and "offscreen" explicitly, then reference them in the property definition. This way you don't have to repeat yourself for the "auto" vs "hidden" spec, and make it easier to read because "offscreen" has an intuitive meaning that everyone should guess at without having to read the exact definition.

Reply:
"skipped" -> add on contain: size, not painted or hit tested [I named it "skipped" to differentiate from "hidden" which is a property value we're defining, as well as a thing elsewhere such as in visibility: hidden.

"offscreen" -> border box does not intersect the viewport plus fixed margin

"has containment" -> add on contain:style layout

Now the definitions:
"visible" -> no change
"auto": has containment, skipped when offscreen
"hidden": has containment, skipped.

Reply:
Makes sense. Done.